test/booster_test.rb

```diff
@@ -88,6 +73,68 @@ def test_predict_rover
     end
   end
 
+  def test_predict_type_leaf_index
+    x_test = [[3.7, 1.2, 7.2, 9.0], [7.5, 0.5, 7.9, 0.0]]
+    leaf_indexes = booster.predict(x_test, predict_type: LightGBM::C_API_PREDICT_LEAF_INDEX)
+    assert_equal 200, leaf_indexes.count
+    assert_equal 9.0, leaf_indexes.first
+    assert_equal 7.0, leaf_indexes.last
+
+    x_test = [3.7, 1.2, 7.2, 9.0]
+    leaf_indexes = booster.predict(x_test, predict_type: LightGBM::C_API_PREDICT_LEAF_INDEX)
+    assert_equal 100, leaf_indexes.count
+    assert_equal 9.0, leaf_indexes.first
+    assert_equal 10.0, leaf_indexes.last
+  end
+
+  def test_predict_type_contrib
+    x_test = [[3.7, 1.2, 7.2, 9.0], [7.5, 0.5, 7.9, 0.0]]
+    results = booster.predict(x_test, predict_type: LightGBM::C_API_PREDICT_CONTRIB)
+    assert_equal 10, results.count
+
+    # split results on num_features + 1
+    predictions = results.each_slice(5).to_a
+    shap_values_1 = predictions.first[0..-2]
+    ypred_1 = predictions.first[-1]
+    assert_elements_in_delta [
+      -0.0733949225678886, -0.24289592050101766, 0.24183795683166504, 0.063430775771174
+    ], shap_values_1
+    assert_in_delta (0.9933333333834246), ypred_1
+
+    shap_values_2 = predictions.last[0..-2]
+    ypred_2 = predictions.last[-1]
+    assert_elements_in_delta [
+      0.1094902954684793, -0.2810485083947154, 0.26691627597706397, -0.13037702397316747
+    ], shap_values_2
+    assert_in_delta (0.9933333333834246), ypred_2
```

Comment:
I don't understand why the last value (supposed to be the prediction result) is the same on each row, and why different from expected `ypred` values in other tests.